lib/block-supports/elements.php

```diff
@@ -200,11 +189,43 @@ function gutenberg_render_elements_support_styles( $pre_render, $block ) {
 		}
 	}
 
-	return null;
+	return $parsed_block;
+}
+
+/**
+ * Ensure the elements block support class name generated and added to
+ * block attributes in the `render_block_data` filter gets applied to the
+ * block's markup.
+ *
+ * @see gutenberg_render_elements_support_styles
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ *
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_elements_class_name( $block_content, $block ) {
+	$class_string = $block['attrs']['className'] ?? '';
+	preg_match( '/\bwp-elements-\S+\b/', $class_string, $matches );
```

Comment:
I'm not sure this will correctly match classname where the target classname is the only or last item in the classlist.


<img width="584" alt="Screenshot 2024-03-04 at 2 20 38 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/4481996a-12a5-4bfd-98c9-d7cc406b85a3">

Maybe a lookbehind?

`/(?<=["\s])(wp-elements-\w+)/`

Reply:
I thought it was safe in this case because it's only looking at the `className` attribute in `$block['attrs']`, and not the class name within an HTML string?

Reply:
Ignore me then. 😄

Reply:
I might be missing something. The `$block['atts']['className']` value isn't HTML markup. It would only be `test wp-elements-1as32dfdfdf`. The `\b` word bound still looks like it should be ok in the tests I've done.

<img width="522" alt="Screenshot 2024-03-04 at 1 31 09 pm" src="https://github.com/WordPress/gutenberg/assets/60436221/79ff58a5-c3ad-4ee1-af45-c1f7e56430a2">

<img width="323" alt="Screenshot 2024-03-04 at 1 31 12 pm" src="https://github.com/WordPress/gutenberg/assets/60436221/268b1cea-f3eb-455a-8656-bcd0c77610e4">

Reply:
I resolved this comment already. Thanks for checking. I didn't read it right.